src/foam/i18n/LocaleTranslationService.js

```diff
@@ -20,8 +29,24 @@ foam.CLASS({
     {
       name: 'getTranslation',
       javaCode: `
-      System.err.println("*********************************** getTranslation");
-      return "foobar";
+        if ( locale.equals("en") ) return originalText;
+
+        DAO localeDAO = (DAO) x.get("localeDAO");
+
+        List<Locale> translationInArray = (ArrayList<Locale>) ((ArraySink) localeDAO.where(
+          AND(
+            EQ(Locale.SOURCE, source),
+            OR(
+              EQ(Locale.LOCALE, locale),
+              EQ(Locale.VARIANT, locale),
+              EQ(Locale.LOCALE_VARIANT, locale)
+            )
+          )
+        ).limit(1).select(new ArraySink())).getArray();
+        
+        String translation = translationInArray.size() > 0 ? translationInArray.get(0).getTarget() : originalText;
```

Comment:
Do we want to return the original english text or an empty string if no translation exists?

Reply:
the original.

Reply:
This is too expensive and it is non-deterministic. If there are two translations, you don't know which one you will get from the limit(1). First look for with variant and then if you don't find look with just the locale.